File: online-judge-solutions/Codeforces/1698B.cpp
//https://codeforces.com/problemset/problem/1698/B
//Lang		:	GNU C++14
//Time		:	343 ms
//Memory	:	800 KB
#include<iostream>
#include<vector>
using namespace std;
int main(){
	int t,n,k;
	cin>>t;
	while(t--){
		cin>>n>>k;
		vector<int> v(n);
		for(int i=0;i<n;i++){cin>>v[i];}
		if(k==1) cout<<(n-1)/2<<endl;
		else{
			int r=0;
			for(int i=1;i<n-1;i++){
				r+=v[i]>v[i-1]+v[i+1];
			}
			cout<<r<<endl;
		}
	}
	return 0;
}

File: online-judge-solutions/Codeforces/1698B.py
#https://codeforces.com/problemset/problem/1698/B
#Lang		:	Python 3.8
#Time		:	186 ms
#Memory         :	15300 KB
for _ in range(int(input())):
    n,k=map(int,input().split())
    v=list(map(int,input().split()))[:n]
    if(k==1):print(int((n-1)/2))
    else:
        r=0
        for i in range(1,n-1):r+=v[i]>v[i-1]+v[i+1]
        print(r)
